src/modules/common/components/TransactionSocketWrapper/index.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+import { toast } from 'react-toastify';
+import client from 'src/utils/api/client';
+import { TRANSACTIONS, AUTH } from 'src/const/queries';
+import { filterIncomingTransactions } from '@transaction/utils/helpers';
+import { fromRawLsk } from '@token/fungible/utils/lsk';
+import i18n from 'src/utils/i18n/i18n';
+import { useCurrentAccount } from '@account/hooks';
+import { useCurrentApplication } from '@blockchainApplication/manage/hooks';
+import { useTransactions, useTransactionsConfig } from '@transaction/hooks/queries';
+
+const showNotificationsForIncomingTransactions = (transactions, account, token) => {
+  filterIncomingTransactions(transactions, account).forEach((transaction) => {
+    const amount = fromRawLsk(transaction.params.amount);
+    if (amount > 0) {
+      const message = transaction.params.data
+        ? i18n.t('with message {{message}}', { message: transaction.params.data })
+        : '';
+      // @TODO: To be fixed in #4652
+      // eslint-disable-next-line no-new
+      new Notification(i18n.t('{{amount}} {{token}} Received', { amount, token }), {
+        body: i18n.t('Your account just received {{amount}} {{token}} {{message}}', {
+          amount,
+          token,
+          message,
+        }),
+      });
+      toast.info(
+        i18n.t('Your account just received {{amount}} {{token}} {{message}}', {
+          amount,
+          token,
+          message,
+        })
+      );
+    }
+  });
+};
+
+const TransactionSocketWrapper = () => {
+  const queryClient = useQueryClient();
+  const [currentAccount] = useCurrentAccount();
+  const [{ chainID }] = useCurrentApplication();
+  const queryConfig = useTransactionsConfig()();
+  const transactionQueryKeys = [TRANSACTIONS, chainID, queryConfig];
+  // Call useTransactions to initialize the cache
+  useTransactions();
+
+  useEffect(() => () => client.socket.off('new.transactions'));
+
+  client.socket.on('new.transactions', async (latestTxns) => {
```

Comment:
What is the reason not to wrap `client.socket.on` in useEffect, wont this create a lot of duplication by calling `on` on every render? This is a pattern we seem to do with blocks as well. And we are not cleaning up the sockets and therefore get a lot of chrome warnings regarding memory leak (although that warning is deprecated in latest react i think it might be valid in this case)

Reply:
I think we need to do this for our useBlocks as well. To prevent xxx number of duplicate listeners 
<img width="360" alt="image" src="https://user-images.githubusercontent.com/8784876/206162232-b8d3f057-34b5-4adf-8246-edf22b209c97.png">

Hook implementation from the [docs](https://socket.io/how-to/use-with-react-hooks). 
Although this listener (new.transactions) managed much better than the other ones, i think it is because we do socket.off in the end of this listener.

Reply:
@oskarleonard Fixed